### PR TITLE
Bug 1862979: daemon: fallback to podman copy when oc fails to extract image

### PR DESCRIPTION
**Pull request by sinnykumari**
This PR fixes:
1. Fallback to podman copy when oc fails to extract image
2. Fallback to podman inspect if imageInspect doesn't work
2. Reduce oc and imageInspect retry to 2. Time to retry grows exponentially, so let's keep a balance.

### CODE REVIEW

**Comment by sinnykumari** [comment]
Keeping early work here for easier testing with clusterbot and ci

**Comment by runcom** [comment]
cc @openshift/openshift-team-mco-maintainers @cgwalters

**Comment by runcom** [comment]
firstboot seems to have failed from log-bundle in e2e-gcp-op

```
Aug 04 14:51:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: error: failed to update OS to registry.svc.ci.openshift.org/ci-op-5d23lkiq/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7 : exit status 1
```

**Comment by runcom** [comment]
not sure _how_ it failed tho:
```
-- Logs begin at Tue 2020-08-04 14:45:12 UTC, end at Tue 2020-08-04 15:20:31 UTC. --
Aug 04 14:50:44 ci-op-5d23lkiq-1354f-b744s-master-2 systemd[1]: Starting Machine Config Daemon Firstboot...
Aug 04 14:50:44 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:50:44.874931    2121 rpm-ostree.go:196] Running captured: rpm-ostree status --json
Aug 04 14:50:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:50:45.152796    2121 daemon.go:222] Booted osImageURL:  (46.82.202007212240-0)
Aug 04 14:50:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:50:45.244341    2121 daemon.go:229] Installed Ignition binary version: 2.3.0
Aug 04 14:50:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:50:45.245922    2121 update.go:381] Checking Reconcilable for config mco-empty-mc to rendered-master-bfc6bd9ed122ed22a00cc85286fbf19d
Aug 04 14:50:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:50:45.246642    2121 update.go:1493] Starting update from mco-empty-mc to rendered-master-bfc6bd9ed122ed22a00cc85286fbf19d: &{osUpdate:true kargs:false fips:false passwd:false files:false units:false kernelType:false extensions:false}
Aug 04 14:50:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:50:45.250966    2121 update.go:921] Updating files
Aug 04 14:50:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:50:45.251354    2121 update.go:958] Deleting stale data
Aug 04 14:50:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:50:45.254802    2121 update.go:234] Falling back to using podman cp to fetch OS image content
Aug 04 14:50:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:50:45.254916    2121 run.go:18] Running: podman pull -q --authfile /var/lib/kubelet/config.json registry.svc.ci.openshift.org/ci-op-5d23lkiq/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7
Aug 04 14:51:26 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: 55bfff42c312e4320a97127254e147aaa6aaca590af0b8360d00719123f2f5a7
Aug 04 14:51:26 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:51:26.739508    2121 rpm-ostree.go:196] Running captured: podman create --net=none --annotation=org.openshift.machineconfigoperator.pivot=true --name ostree-container-pivot-372bd17c-4171-4d7c-8b23-010de7000aeb registry.svc.ci.openshift.org/ci-op-5d23lkiq/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7
Aug 04 14:51:26 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:51:26.883466    2121 run.go:18] Running: podman cp d45fa653c247bd8bfedab6250748507155fec24fc47046ed2a0ab0288f4cedfb:/ /run/mco-machine-os-content/os-content-197556798
Aug 04 14:51:44 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:51:44.422544    2121 update.go:261] Deleted container and removed OSContainer image
Aug 04 14:51:44 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:51:44.426416    2121 update.go:1371] Updating OS to registry.svc.ci.openshift.org/ci-op-5d23lkiq/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7
Aug 04 14:51:44 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:51:44.426449    2121 rpm-ostree.go:196] Running captured: rpm-ostree status --json
Aug 04 14:51:44 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:51:44.470011    2121 rpm-ostree.go:130] Current origin is not custom
Aug 04 14:51:44 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:51:44.736937    2121 rpm-ostree.go:146] Pivoting to: 46.82.202008032211-0 (ee92a4d447b26264cb0c2499c52a7b78796fbf0a023a8d36ec8aadceedfc40b8)
Aug 04 14:51:44 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:51:44.736978    2121 rpm-ostree.go:178] Executing rebase from pivot://registry.svc.ci.openshift.org/ci-op-5d23lkiq/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7 and checksum ee92a4d447b26264cb0c2499c52a7b78796fbf0a023a8d36ec8aadceedfc40b8
Aug 04 14:51:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:51:45.145593    2121 update.go:921] Updating files
Aug 04 14:51:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: I0804 14:51:45.146037    2121 update.go:958] Deleting stale data
Aug 04 14:51:45 ci-op-5d23lkiq-1354f-b744s-master-2 machine-config-daemon[2121]: error: failed to update OS to registry.svc.ci.openshift.org/ci-op-5d23lkiq/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7 : exit status 1
Aug 04 14:51:45 ci-op-5d23lkiq-1354f-b744s-master-2 systemd[1]: machine-config-daemon-firstboot.service: Main process exited, code=exited, status=1/FAILURE
Aug 04 14:51:45 ci-op-5d23lkiq-1354f-b744s-master-2 systemd[1]: machine-config-daemon-firstboot.service: Failed with result 'exit-code'.
Aug 04 14:51:45 ci-op-5d23lkiq-1354f-b744s-master-2 systemd[1]: Failed to start Machine Config Daemon Firstboot.
Aug 04 14:51:45 ci-op-5d23lkiq-1354f-b744s-master-2 systemd[1]: machine-config-daemon-firstboot.service: Consumed 1min 35.508s CPU time
```

**Comment by runcom** [comment]
ok, this piece failed:
```
	// RPM-OSTree can now directly slurp from the mounted container!
	// https://github.com/projectatomic/rpm-ostree/pull/1732
	err = exec.Command("rpm-ostree", "rebase", "--experimental",
		fmt.Sprintf("%s:%s", repo, ostreeCsum),
		"--custom-origin-url", customURL,
		"--custom-origin-description", "Managed by machine-config-operator").Run()
	if err != nil {
		return
	}
```
w/o logs tho

**Comment by runcom** [comment]
pushed a commit to grab additional logs

**Comment by sinnykumari** [comment]
yeah, default error message is not useful. Thanks Antonio for adding up logs

**Comment by sinnykumari** [comment]
Hmm, not much useful information from rpm-ostree:
```
Aug 04 15:56:18 ci-op-ywjvywbj-1354f-2nhgx-master-0 machine-config-daemon[2106]: I0804 15:56:18.476291    2106 rpm-ostree.go:146] Pivoting to: 46.82.202008032211-0 (ee92a4d447b26264cb0c2499c52a7b78796fbf0a023a8d36ec8aadceedfc40b8)
Aug 04 15:56:18 ci-op-ywjvywbj-1354f-2nhgx-master-0 machine-config-daemon[2106]: I0804 15:56:18.476323    2106 rpm-ostree.go:178] Executing rebase from pivot://registry.svc.ci.openshift.org/ci-op-ywjvywbj/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7 and checksum ee92a4d447b26264cb0c2499c52a7b78796fbf0a023a8d36ec8aadceedfc40b8
Aug 04 15:56:18 ci-op-ywjvywbj-1354f-2nhgx-master-0 machine-config-daemon[2106]: I0804 15:56:18.875689    2106 update.go:921] Updating files
Aug 04 15:56:18 ci-op-ywjvywbj-1354f-2nhgx-master-0 machine-config-daemon[2106]: I0804 15:56:18.876275    2106 update.go:958] Deleting stale data

Aug 04 15:56:18 ci-op-ywjvywbj-1354f-2nhgx-master-0 machine-config-daemon[2106]: error: failed to update OS to registry.svc.ci.openshift.org/ci-op-ywjvywbj/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7 : failed to run rpm-ostree rebase: error: The connection is closed
Aug 04 15:56:18 ci-op-ywjvywbj-1354f-2nhgx-master-0 machine-config-daemon[2106]: : exit status 1
```

**Comment by runcom** [comment]
uhm:
```
Aug 04 15:56:18 ci-op-ywjvywbj-1354f-2nhgx-master-0 machine-config-daemon[2106]: error: failed to update OS to registry.svc.ci.openshift.org/ci-op-ywjvywbj/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7 : failed to run rpm-ostree rebase: error: The connection is closed
```
maybe @cgwalters @jlebon can shed some light on the error

**Comment by sinnykumari** [comment]
added more logs to check that we are getting the right contents and permissions on the files. It would have been easier to debug on a local cluster but so far getting inaccessible master nodes

**Comment by cgwalters** [comment]
> Aug 04 15:56:18 ci-op-ywjvywbj-1354f-2nhgx-master-0 machine-config-daemon[2106]: error: failed to update OS to registry.svc.ci.openshift.org/ci-op-ywjvywbj/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7 : failed to run rpm-ostree rebase: error: The connection is closed

That is weird; I think that error comes from GDBus.  My offhand guess here is that somehow we're rebooting the system while trying to run this code.

**Comment by cgwalters** [comment]
Another potential problem here is to verify that `imageInspect()` is respecting ICSP - I wouldn't be surprised if it wasn't.

I think it would be more robust to drop that function and instead have the oscontainer download process extract it - e.g. if we use `podman pull` then we can get that data from `podman inspect` and pass it along.

File: pkg/daemon/update.go [hunk 1]
```diff
@@ -217,13 +219,62 @@ func addExtensionsRepo(osImageContentDir string) error {
...
+	var authArgs []string
+	if _, err := os.Stat(kubeletAuthFile); err == nil {
+		authArgs = append(authArgs, "--authfile", kubeletAuthFile)
+	}
+	args := []string{"pull", "-q"}
+	args = append(args, authArgs...)
+	args = append(args, imgURL)
+	_, err = pivotutils.RunExt(numRetriesNetCommands, "podman", args...)
+	if err != nil {
+		return
```
Comment (kikisdeliveryservice) [inline pkg/daemon/update.go:246]: this will return err?

**Comment by runcom** [comment]
> Another potential problem here is to verify that `imageInspect()` is respecting ICSP - I wouldn't be surprised if it wasn't.

maybe there's some ICSP special handling that we need to do? I thought it was transparent to containers/image (but I'm likely wrong)

**Comment by jlebon** [comment]
> ```
> Aug 04 15:56:18 ci-op-ywjvywbj-1354f-2nhgx-master-0 machine-config-daemon[2106]: error: failed to update OS to registry.svc.ci.openshift.org/ci-op-ywjvywbj/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7 : failed to run rpm-ostree rebase: error: The connection is closed
> ```

This could also be SELinux issues. Maybe it allows D-Bus passing the repo fd to rpm-ostreed if it has a container context, but not if it's in `/run`? As a hack, you can try adding `chcon -t container_var_lib_t /path/to/repo` before the rebase.

File: pkg/daemon/update.go (hunk 1 above)
Comment (sinnykumari) [reply pkg/daemon/update.go:246]: yes, it is named return https://tour.golang.org/basics/7

**Comment by sinnykumari** [comment]
> > ```
> > Aug 04 15:56:18 ci-op-ywjvywbj-1354f-2nhgx-master-0 machine-config-daemon[2106]: error: failed to update OS to registry.svc.ci.openshift.org/ci-op-ywjvywbj/stable@sha256:16a8dde4b893ff0b5b4aeb05474f2f5e2ce9cac45d5d3e98b40c4309e23215a7 : failed to run rpm-ostree rebase: error: The connection is closed
> > ```
> 
> This could also be SELinux issues. Maybe it allows D-Bus passing the repo fd to rpm-ostreed if it has a container context, but not if it's in `/run`? As a hack, you can try adding `chcon -t container_var_lib_t /path/to/repo` before the rebase.

Not sure if this is really a problem. Before implementing I tested the steps on a local cluster pod with all these steps `podma pull && podman create && podman copy /run/mco-machine-os-content/os-content-XXX` and ran ostree rebase which went fine.

**Comment by sinnykumari** [comment]
Ok, so it is **selinux**
```
type=AVC msg=audit(1596608327.782:158): avc:  denied  { read } for  pid=1417 comm="dbus-daemon" path="/run/mco-machine-os-content/os-content-060048094/srv/repo" dev="tmpfs" ino=136993 scontext=system_u:system_r:system_dbusd_t:s0-s0:c0.c1023 tcontext=unconfined_u:object_r:container_var_run_t:s0 tclass=dir permissive=0
```

From the log, default selinux context set on the repo is
```
system_u:object_r:container_var_run_t:s0
```

selinux denial is still there when we set type to container_var_lib_t and rebase

$ sudo chcon -R -t container_var_lib_t /run/mco-machine-os-content/os-content-060048094/
```
type=AVC msg=audit(1596608928.864:281): avc:  denied  { read } for  pid=1417 comm="dbus-daemon" path="/run/mco-machine-os-content/os-content-060048094/srv/repo" dev="tmpfs" ino=136993 scontext=system_u:system_r:system_dbusd_t:s0-s0:c0.c1023 tcontext=unconfined_u:object_r:container_var_lib_t:s0 tclass=dir permissive=0
```

**Comment by sinnykumari** [comment]
After doing some comparison with what selinux context is set when doing podman cp from container and from host, I see that in container context it sets `var_run_t` while running directly on host it sets `container_var_run_t` . Since, var_run_t context was working for me locally from a pod, updating context to `var_run_t` on host working as well locally. Updated PR to see the result.

I hope I am making sense :D

**Comment by runcom** [comment]
oh well, op passed :)

**Comment by runcom** [comment]
let's check metal even if I think they're using ipv4 and connected nowadays

**Comment by sinnykumari** [comment]
I didn't find any useful log in metal-ipi test log. Retrying though
/test e2e-metal-ipi

**Comment by sinnykumari** [comment]
> Another potential problem here is to verify that `imageInspect()` is respecting ICSP - I wouldn't be surprised if it wasn't.
> 
> I think it would be more robust to drop that function and instead have the oscontainer download process extract it - e.g. if we use `podman pull` then we can get that data from `podman inspect` and pass it along.

We wanted to first check if  imageInspect()  works or not. If it doesn't we can use podman inspect to get ostree infomation for failing cases. We can't generalize podman inspect because the happy path is to use `oc extract`

**Comment by runcom** [comment]
> /test e2e-aws-proxy

not even sure we have this actually

**Comment by runcom** [comment]
alrighty, I think we can proceed with this since the BM team is going through another (not related) issue [1] and can't fully test our fix out just yet (moved BZ to urgent).
However since we know we broke ICSP, we've been testing Sinny's PR with mirror and it does work + testing 4.6.nightly would result in an `oc image extract` error (this time, on gcp resulting in an image not found vs forbidden, https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-launch-gcp/1291060829621325824).

Just a tiny note: https://bugzilla.redhat.com/show_bug.cgi?id=1862948#c3 <- this BZ seems to be also related to `oc` but talks about proxy, we're investigating that separately and we'd want to merge this to unblock disconnected+ipv6 CI once that other network fix is done. This shouldn't impact _our_ testing as confirmed by _our_ CI also.

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1866464

/approve

File: pkg/daemon/rpm-ostree.go [hunk 2]
```diff
@@ -131,18 +185,29 @@ func (r *RpmOstreeClient) Rebase(container, osImageContentDir string) (changed b
...
+			}
+			ostreeCsum = podmanImgData.Labels["com.coreos.ostree-commit"]
+			ostreeVersion = podmanImgData.Labels["version"]
+		}
+	} else {
+		ostreeCsum = imageData.Labels["com.coreos.ostree-commit"]
+		ostreeVersion = imageData.Labels["version"]
 	}
+	// We may have pulled in OSContainer image as fallback during podmanCopy() or podmanInspect()
+	defer exec.Command("podman", "rmi", imgURL).Run()
```
Comment (runcom) [inline pkg/daemon/rpm-ostree.go:203]: should this be `podman rmi -f` to make it idempotent in case we haven't pulled it?

**Comment by runcom** [comment]
did a couple of other jobs with mirrors both with this PR and without and I'm getting the same result, works with this PR, doesn't even install w/o

File: pkg/daemon/update.go [hunk 3]
```diff
@@ -217,6 +219,56 @@ func addExtensionsRepo(osImageContentDir string) error {
...
+
+	// only delete created container, we will delete container image later as we may need it for podmanInspect()
+	defer podmanRemove(containerName)
+
+	// copy the content from create container locally into a temp directory under /run/machine-os-content/
+	cid := strings.TrimSpace(string(cidBuf))
+	args = []string{"cp", fmt.Sprintf("%s:/", cid), osImageContentDir}
+	_, err = pivotutils.RunExt(numRetriesNetCommands, "podman", args...)
+
+	// Set selinux context to var_run_t to avoid selinux denial
```
Comment (cgwalters) [inline pkg/daemon/update.go:262]: One random aside: SELinux is really good about protecting the host from containers, and avoiding containers interacting with other containers that they shouldn't.

But, we're trying to cross these two worlds, and that's the source of pain.

**Comment by kikisdeliveryservice** [comment]
> > /test e2e-aws-proxy
> 
> ~not even sure we have this actually~ oh yeah we have

does this need to pass @runcom ?

**Comment by kikisdeliveryservice** [comment]
actually job is red for every run..ever?: https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/pull-ci-openshift-machine-config-operator-master-e2e-aws-proxy

File: pkg/daemon/update.go (hunk 3 above)
Comment (sinnykumari) [reply pkg/daemon/update.go:262]: yeah! we have come long way forward with running everything in container context. With always improvement in these awesome software, we can hope to reduce the pain in future.

File: pkg/daemon/rpm-ostree.go (hunk 2 above)
Comment (sinnykumari) [reply pkg/daemon/rpm-ostree.go:203]: +1, forcing is good to do here.

**Comment by sinnykumari** [comment]
I have added additional fixes to reduce oc and imageInspect retry to 2. Time to retry oc and imageInspect on failure grows exponentially. As we know there are chances of these commands failing in certain environment, these retry can add up additional time to update.

Although there is possibility that in certain cases due to network glitch these commands fails which can be fixed with
a retry.

Keeping retry to 2 keeps a balance between both scenarios.

**Comment by runcom** [comment]
> actually job is red for every run..ever?: https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/pull-ci-openshift-machine-config-operator-master-e2e-aws-proxy

+1 I'll create a BZ to track that for us (and follow up for https://bugzilla.redhat.com/show_bug.cgi?id=1862948#c3) - the thing with that job is that I couldn't find any install gather, we can sync with @danehans perhaps to understand it better and fix it

**Comment by sinnykumari** [comment]
@vrutkovs any idea why okd-e2e-aws is failing on this PR? Can you confirm that it not because of `oc` being not available in OKD.

**Comment by vrutkovs** [comment]
OKD tests are broken due to installer / CEO changes :/ Lets ignore these results for now

**Comment by ashcrow** [comment]
We have 2 positive reviews and all requested feedback has been incorporated. The related tests that failed are either known to be broken or are breaking on unrelated items.

/lgtm

**Comment by runcom** [comment]
> actually job is red for every run..ever?: https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/pull-ci-openshift-machine-config-operator-master-e2e-aws-proxy

@kikisdeliveryservice I think MCO doesn't even have support for e2e-proxy in openshift/release - could that be the case?

**Comment by kikisdeliveryservice** [comment]
test was added here: https://github.com/openshift/machine-config-operator/pull/1979

**Comment by sinnykumari** [comment]
> test was added here: #1979

It points to the very same PR :)

**Comment by runcom** [comment]
> > test was added here: #1979
> 
> It points to the very same PR :)

inception

**Comment by runcom** [comment]
we spoke on slack as well - the openshift/release PR is https://github.com/openshift/release/pull/10380 (thanks Kirsten!)

**Comment by kikisdeliveryservice** [comment]
> > > test was added here: #1979
> > 
> > 
> > It points to the very same PR :)
> 
> inception

That was truly copypasta. :laughing:

**Comment by sinnykumari** [comment]
This PR seems to be working well in proxy setup. For reference, here is log from one of the test cluster:
```
masters-journal:Aug 07 07:33:05.936202 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: error: image "registry.svc.ci.openshift.org/ci-ln-4tzw542/release-scratch@sha256:9ebb2055da173f49371dbddee642d59bd677bc3658ed4ce4b11f3672f11979e1" does not exist
masters-journal:Aug 07 07:33:05.938356 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: W0807 07:33:05.938313    2119 run.go:44] oc failed: running oc image extract --path /:/run/mco-machine-os-content/os-content-757029966 --registry-config /var/lib/kubelet/config.json registry.svc.ci.openshift.org/ci-ln-4tzw542/release-scratch@sha256:9ebb2055da173f49371dbddee642d59bd677bc3658ed4ce4b11f3672f11979e1 failed: error: image "registry.svc.ci.openshift.org/ci-ln-4tzw542/release-scratch@sha256:9ebb2055da173f49371dbddee642d59bd677bc3658ed4ce4b11f3672f11979e1" does not exist
masters-journal:Aug 07 07:33:05.938356 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: : exit status 1; retrying...
masters-journal:Aug 07 07:33:05.938529 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:05.938392    2119 update.go:300] Falling back to using podman cp to fetch OS image content
masters-journal:Aug 07 07:33:05.938529 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:05.938440    2119 run.go:18] Running: podman pull -q --authfile /var/lib/kubelet/config.json registry.svc.ci.openshift.org/ci-ln-4tzw542/release-scratch@sha256:9ebb2055da173f49371dbddee642d59bd677bc3658ed4ce4b11f3672f11979e1
masters-journal:Aug 07 07:33:38.797990 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: 4986f8f9d0a648aa8f36d2d763da618eed9256a52be3a74c15fdfdda127601f3
masters-journal:Aug 07 07:33:38.812308 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:38.812250    2119 rpm-ostree.go:261] Running captured: podman create --net=none --annotation=org.openshift.machineconfigoperator.pivot=true --name ostree-container-pivot-b20d11a8-a4e8-48b2-979a-74e5bb6da76c registry.svc.ci.openshift.org/ci-ln-4tzw542/release-scratch@sha256:9ebb2055da173f49371dbddee642d59bd677bc3658ed4ce4b11f3672f11979e1
masters-journal:Aug 07 07:33:38.970258 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:38.970205    2119 run.go:18] Running: podman cp 2c738a24b4ca8aed66cb7362175555773611795e82d347af4075c877e5c89ee4:/ /run/mco-machine-os-content/os-content-757029966
masters-journal:Aug 07 07:33:56.182259 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:56.182207    2119 rpm-ostree.go:261] Running captured: chcon -R -t var_run_t /run/mco-machine-os-content/os-content-757029966
masters-journal:Aug 07 07:33:57.013202 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:57.013163    2119 update.go:1372] Updating OS to registry.svc.ci.openshift.org/ci-ln-4tzw542/release-scratch@sha256:9ebb2055da173f49371dbddee642d59bd677bc3658ed4ce4b11f3672f11979e1
masters-journal:Aug 07 07:33:57.013202 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:57.013163    2119 update.go:1372] Updating OS to registry.svc.ci.openshift.org/ci-ln-4tzw542/release-scratch@sha256:9ebb2055da173f49371dbddee642d59bd677bc3658ed4ce4b11f3672f11979e1
masters-journal:Aug 07 07:33:57.013202 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:57.013195    2119 rpm-ostree.go:261] Running captured: rpm-ostree status --json
masters-journal:Aug 07 07:33:57.090834 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:57.090784    2119 rpm-ostree.go:184] Current origin is not custom
masters-journal:Aug 07 07:33:57.288992 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:57.288801    2119 rpm-ostree.go:211] Pivoting to: 46.82.202008062240-0 (fb328535bd11ca43d4d7cd75bd16dcedaa38a3b71a5eab588d1f143bc958cf98)
masters-journal:Aug 07 07:33:57.288992 ci-ln-4tzw542-f76d1-2gl4k-master-0 machine-config-daemon[2119]: I0807 07:33:57.288840    2119 rpm-ostree.go:261] Running captured: rpm-ostree rebase --experimental /run/mco-machine-os-content/os-content-757029966/srv/repo:fb328535bd11ca43d4d7cd75bd16dcedaa38a3b71a5eab588d1f143bc958cf98 --custom-origin-url pivot://registry.svc.ci.openshift.org/ci-ln-4tzw542/release-scratch@sha256:9ebb2055da173f49371dbddee642d59bd677bc3658ed4ce4b11f3672f11979e1 --custom-origin-description Managed by machine-config-operator
```

**Pull request merged**